### DISCUSSION TITLE
Upgrade librdkafka ?

### DISCUSSION THREAD

**Discussion by gfeyer** [microsoft/vcpkg#17328] [Ideas]
Is there a way to upgrade librdkafka to the latest version ? 

There are many bug fixes in the lates version in github vs vcpkg version of 1.5.0 

Thanks,
Gabriel.

**Comment by mathisloge** [comment]
If you have some spare time, you could open a PR and update to the latest release https://github.com/edenhill/librdkafka/releases/tag/v1.6.1 . 

A good start is https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md .

**Reply by gfeyer** [reply]
Thanks for the documentation, I gave that a read but still unsure how to just upgrade the tag
For example, in this file
[librdkafka/portfile.cmake](https://github.com/microsoft/vcpkg/blob/master/ports/librdkafka/portfile.cmake)

I'd just need to update the REF and SHA512 from vcpkg_from_github(..)? I've looked at the other files and portfile.cmake seems the most appropriate to use the new 1.6.1 tag. 

Thanks for your time
Gabriel

**Reply by mathisloge** [reply]
generally the first steps are to change the version in the portfile as you described and also change the version in the vcpkg.json. 
(Convert the CONTROL file with `vcpkg format-manifest ports\librdkafka\CONTROL` to a manifest file (vcpkg.json))

To help you get started, I have made the basic changes https://github.com/mathisloge/vcpkg/commit/4eef67bb22650fdbe07d091bc1ac7fba3fde4287 here. Additionally you will need to change `version-string` to `version-semver`. 

*After* you have commited anything which has changes inside port/librdkafka you have to update the versions. 
`vcpkg x-add-version librdkafka` 

You can read more about the manifest files here https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md 

After you have applied the changes in the mentioned commit and steps described here, just open a PR. Optimally you can test with a external application, that nothing breaks with this update. The CI testing the PR will also test other platforms.
